### DISCUSSION TITLE
Trouble including dependency from eclipse target platform

### DISCUSSION THREAD

**Discussion by tcornerstone** [eclipse-tycho/tycho#4354] [General]
Hello, I'm new to using Tycho and want to mavenize an existing Eclipse plugin using Tycho. The project uses only Eclipse as target platform, which I am trying to set up like described in the wiki. However, when building the project, Tycho can't locate any dependencies. Even though index files are downloaded from the repository, the resulting error messages are the same as if no target platform was specified at all:

```
[INFO] --------------------------[ eclipse-feature ]---------------------------
[INFO] Resolving dependencies of MavenProject: com.example:com.example.feature:1.0.0-SNAPSHOT @ /home/user/tycho-example/features/com.example.feature/.polyglot.feature.xml
[INFO] ### Using TychoRepositoryTransport for remote P2 access ###
[INFO]     Cache location:         /home/user/.m2/repository/.cache/tycho
[INFO]     Transport mode:         online
[INFO]     Http Transport type:    Java11Client
[INFO]     Update mode:            cache first
[INFO]     Minimum cache duration: 60 minutes
[INFO]       (you can configure this with -Dtycho.p2.transport.min-cache-minutes=<desired minimum cache duration>)
[INFO] Downloading from p2: http://download.eclipse.org/releases/2023-12/p2.index
[INFO] Downloaded from p2: http://download.eclipse.org/releases/2023-12/p2.index (122 bytes at 0 bytes/s)
[INFO] Adding repository http://download.eclipse.org/releases/2023-12
[INFO] Downloading from p2: http://download.eclipse.org/technology/epp/packages/2023-12/p2.index
[INFO] Downloading from p2: http://download.eclipse.org/technology/epp/packages/2023-12/202312011200/p2.index
[INFO] Downloading from p2: http://download.eclipse.org/releases/2023-12/202312061001/p2.index
[INFO] Downloaded from p2: http://download.eclipse.org/releases/2023-12/202312061001/p2.index (140 bytes at 0 bytes/s)
[INFO] Downloading from p2: http://download.eclipse.org/releases/2023-12/p2.index
[INFO] Downloaded from p2: http://download.eclipse.org/releases/2023-12/p2.index (122 bytes at 119 KB/s)
[INFO] Downloading from p2: http://download.eclipse.org/technology/epp/packages/2023-12/p2.index
[INFO] Downloading from p2: http://download.eclipse.org/technology/epp/packages/2023-12/202312011200/p2.index
[INFO] Downloading from p2: http://download.eclipse.org/releases/2023-12/202312061001/p2.index
[INFO] Downloaded from p2: http://download.eclipse.org/releases/2023-12/202312061001/p2.index (140 bytes at 136 KB/s)
[INFO] Resolving dependencies of MavenProject: com.example:com.example.feature:1.0.0-SNAPSHOT @ /home/teckste/tycho-example/features/com.example.feature/.polyglot.feature.xml
[INFO] {osgi.os=linux, osgi.ws=gtk, org.eclipse.update.install.features=true, osgi.arch=x86_64, org.eclipse.update.install.sources=true}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: com.example.feature.feature.group 1.0.0.qualifier
[ERROR]   Missing requirement: com.example.feature.feature.group 1.0.0.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.debug.ui.launchview 0.0.0' but it could not be found
```

For the sake of demonstration, I have created a minimal example based on the project I'm working on where the error occurs. It consists of a single bundle with one requirement: [tycho-example.zip](https://github.com/user-attachments/files/17361755/tycho-example.zip)

I'm not sure whether I'm doing something wrong or there is a bug in Tycho. I suspect the former.

**Comment by laeubi** [comment]
> For the sake of demonstration, I have created a minimal example based on the project I'm working on where the error occurs. It consists of a single bundle with one requirement: [tycho-example.zip](https://github.com/user-attachments/files/17361755/tycho-example.zip)

Your example only consist of one feature (not a bundle). Beside that you don't seem to use a target file what I would recommend to get the same behavior in the IDE and the build. Then make sure the IU `org.eclipse.debug.ui.launchview`  is really part of the target platform, also using latest Tycho  + Eclipse releases.

**Comment by tcornerstone** [comment]
Specifying a target platform file did work eventually, but given that including a target platform via URL seems to be completely broken, why is it documented in the wiki?
[https://wiki.eclipse.org/Tycho/Target_Platform/#Simple_target_platform_configuration](https://wiki.eclipse.org/Tycho/Target_Platform/#Simple_target_platform_configuration)

**Reply by laeubi** [reply]
It works perfectly fine for most of us, I just can give you some advice on how it might be easier to find out whats wrong here, otherwise you can try to use the repository explorer in eclipse to check if the IU you requested is actually there,
